src/Codeception/Lib/Driver/Db.php

```diff
@@ -153,22 +153,47 @@ public function select($column, $table, array &$criteria)
         return sprintf($query, $column, $this->getQuotedName($table), $where);
     }
 
+    private function getSupportedOperators(){
+      return [
+        'like' => 'LIKE',
+        '<=' => '<=',
+        '>=' => '>=',
+        '<' => '<',
+        '>' => '>'
+      ];
+    }
+
     protected function generateWhereClause(array &$criteria)
     {
         if (empty($criteria)) {
             return '';
         }
 
+        $operands = $this->getSupportedOperators();
+
         $params = [];
         foreach ($criteria as $k => $v) {
-            if ($v === null) {
-                $params[] = $this->getQuotedName($k) . " IS NULL ";
-                unset($criteria[$k]);
-            } elseif (strpos(strtolower($k), ' like') > 0) {
-                $k = str_replace(' like', '', strtolower($k));
-                $params[] = $this->getQuotedName($k) . " LIKE ? ";
-            } else {
-                $params[] = $this->getQuotedName($k) . " = ? ";
+            if ($v === null)
+            {
+              $params[] = $this->getQuotedName( $k ) . " IS NULL ";
+              unset( $criteria[ $k ] );
+              continue;
+            }
+
+            $bHasOperand = false;
```

Comment:
this code is pretty complicated for the first sight to understand and debug. 
could you please add some comments to it